waspc/src/Wasp/Generator/JsImport.hs

```diff
@@ -1,26 +1,34 @@
 module Wasp.Generator.JsImport
-  ( getJsImportDetailsForExtFnImport,
+  ( mkJsImportGetterFromExtSrcDir,
+    RelDirToExternalCodeDir,
   )
 where
 
-import StrongPath (Dir, Path, Posix, Rel, (</>))
+import StrongPath (Dir, Dir', Path, Posix, Rel, (</>))
 import qualified StrongPath as SP
-import qualified Wasp.AppSpec.ExtImport as AS.ExtImport
+import qualified Wasp.AppSpec.ExtImport as EI
 import Wasp.Generator.ExternalCodeGenerator.Common (GeneratedExternalCodeDir)
+import Wasp.JsImport
+  ( JsImport,
+    JsImportName (JsImportField, JsImportModule),
+    getJsImport,
+  )
+
+type RelDirToExternalCodeDir = Path Posix (Rel GeneratedExternalCodeDir) Dir'
 
-getJsImportDetailsForExtFnImport ::
-  -- | Path to generated external code directory, relative to the directory in which file doing the importing is.
-  Path Posix (Rel a) (Dir GeneratedExternalCodeDir) ->
-  AS.ExtImport.ExtImport ->
-  -- | (importIdentifier, importStmt)
-  --   - importIdentifier -> Identifier via which you can access ext js function after you import it with importStmt.
-  --   - importStmt -> Javascript import statement via which you should do the import.
-  (String, String)
-getJsImportDetailsForExtFnImport relPosixPathToExtCodeDir extImport = (importIdentifier, importStmt)
+-- | Returns a getter function that can be used to get a JsImport from an ExtImport
+mkJsImportGetterFromExtSrcDir ::
```

Comment:
I would rename this one to `jsImportFromExtImport` or `extImportToJsImport`. Probably the first one. And would keep its name the same as the functions you have in ServerGenerator and ClientGenerator for doing this same thing. And then I would also remove those extra parentheses in the type definition, although I told you earlier those can make sense -> but I don't think they make sense here. Why? Because instead of looking at it as producing another function, which could be valid but I don't think is the best option in this case, you can look at it as a more general version of those functions you have in Server and Client Generators. Let's call this one A, and we can call those B and C. So B and C are the same function, just in different context, which enables them to just call the A with only first argument, which comes from their context, and the rest are propagated. So B and C are just specializations of A. And therefore it is even ok that they all have the same name, because they have the additional context of module they are imported from.
I think this is the cleanest way do it. This way, function A doesn't even have to know that B and C exist. While it kind of does in your case -> it doesn't know about them particularly, but it does have this notion that it is producing another function that will be used somewhere else, while it doesn't need to know that. Ok I expressed this last part relatively clumsily so take it with a grain of salt.

Reply:
> I would rename this one to `jsImportFromExtImport` or `extImportToJsImport. Probably the first one. And would keep its name the same as the functions you have in ServerGenerator and ClientGenerator for doing this same thing. 
>
> And then I would also remove those extra parentheses in the type definition, although I told you earlier those can make sense -> but I don't think they make sense here. Why? Because instead of looking at it as producing another function, which could be valid but I don't think is the best option in this case, you can look at it as a more general version of those functions you have in Server and Client Generators. 
> 
> <details>
> <summary>Extra info</summary>
> Let's call this one A, and we can call those B and C. So B and C are the same function, just in different context, which enables them to just call the A with only first argument, which comes from their context, and the rest are propagated. So B and C are just specializations of A. And therefore it is even ok that they all have the same name, because they have the additional context of module they are imported from.
> </details>
> 
> I think this is the cleanest way do it. This way, function A doesn't even have to know that B and C exist. While it kind of does in your case -> it doesn't know about them particularly, but it does have this notion that it is producing another function that will be used somewhere else, while it doesn't need to know that. Ok I expressed this last part relatively clumsily so take it with a grain of salt.

Makes sense 👍 general function exists in `Wasp.Generator` context and specialized functions in `Wasp.Generator.ServerGenerator` and `Wasp.Generator.WebAppGenerator` contexts.